Lib/venv/scripts/common/Activate.ps1

```diff
@@ -1,56 +1,229 @@
+
+If the value starts with a `'` or a `"` then the first and last character is
+stripped from the value before being captured.
+
+.Parameter ConfigDir
+Path to the directory that contains the `pyvenv.cfg` file.
+#>
+function Get-PyVenvCfgOverrides(
+    [String]
+    $ConfigDir
+) {
+    Write-Verbose "Given ConfigDir=$ConfigDir, obtain values in pyvenv.cfg"
+
+    # Ensure the file exists, and issue a warning if it doesn't (but still allow the function to continue).
+    $pyvenvConfigPath = Join-Path -Resolve -Path $ConfigDir -ChildPath 'pyvenv.cfg' -ErrorAction Continue
+
+    if ($pyvenvConfigPath) {
+
+        Write-Verbose "File exists, parse `key = value` lines"
+        $pyvenvConfigContent = Get-Content -Path $pyvenvConfigPath
+        $pyvenvConfig = @{ }
+
+        $pyvenvConfigContent | ForEach-Object {
+            $keyval = $PSItem -split "\s*=\s*", 2
+            if ($keyval[0] -and $keyval[1]) {
+                $val = $keyval[1]
+
+                # Remove extraneous quotations around a string value.
+                if ("'""".Contains($val.Substring(0,1))) {
+                    $val = $val.Substring(1, $val.Length - 2)
+                }
+
+                $pyvenvConfig[$keyval[0]] = $val
+                Write-Verbose "Adding Key: '$($keyval[0])'='$val'"
+            }
+        }
+    }
+    return $pyvenvConfig
+}
+
+
+<# Begin Activate script --------------------------------------------------- #>
+
+# Determine the containing directory of this script
+$VenvExecPath = Split-Path -Parent $MyInvocation.MyCommand.Definition
+$VenvExecDir = Get-Item -Path $VenvExecPath
+
+Write-Verbose "Activation script is located in path: '$VenvExecPath'"
+Write-Verbose "VenvExecDir Fullname: '$($VenvExecDir.FullName)"
+Write-Verbose "VenvExecDir Name: '$($VenvExecDir.Name)"
+
+# Set values required in priority: CmdLine, ConfigFile, Default
+# First, get the location of the virtual environment, it might not be
+# VenvExecDir if specified on the command line.
+if ($VenvDir) {
+    Write-Verbose "VenvDir given as parameter, using '$VenvDir' to determine values"
+} else {
+    Write-Verbose "VenvDir not given as a parameter, using parent directory name as VenvDir."
+    $VenvDir = $VenvExecDir.Parent.FullName.TrimEnd("\\/")
+    $VenvDir = $VenvDir.Insert($VenvDir.Length, "/")
+    Write-Verbose "VenvDir=$VenvDir"
+}
+
+# Next, read the `pyvenv.cfg` file to determine any required value such
+# as `prompt`.
+$pyvenvCfg = Get-PyVenvCfgOverrides -ConfigDir $VenvDir
+
+# Next, set the prompt from the command line, or the config file, or
+# just use the name of the virtual environment folder.
+if ($Prompt) {
+    Write-Verbose "Prompt specified as argument, using '$Prompt'"
+} else {
+    Write-Verbose "Prompt not specified as argument to script, checking pyvenv.cfg value"
+    if ($pyvenvCfg -and $pyvenvCfg['prompt']) {
+        Write-Verbose "  Setting based on value in pyvenv.cfg='$($pyvenvCfg['prompt'])'"
+        $Prompt = $pyvenvCfg['prompt'];
```

Comment:
This is the line that gave me the most nightmares when thinking about how to do this, because the prompt value in the config file is quoted and escaped (basically, it's `repr(prompt)`), so we need to do more processing here.

Reply:
Thanks for the review @zooba!

Agreed that the escaped character problem is a bit tricky in the context of Powershell... `\n` is not really available in Powershell, for instance, but could be replaced with the back-tick-quote `. Finding and replacing _all types_ of escaped characters (newlines, tabs, colors?) might be a bit of a deep dark hole (there be ANSI-escaped coloured dragons there)!

Would it be appropriate to add handling of escaped characters in a different issue/PR? I don't expect that the prior version of this Activate.ps1 script handled this problem particularly well either (a brief test with the latest 3.8 beta shows a few hiccups right away).
Of course, I would be most happy to tackle that problem next!

What sample input would show the shortcoming of this implementation, so that I can write some unit tests and fix it correctly?

Just from thinking about it briefly I can think up input that contains:

- newlines (`\n`, `\r\n`)
- tabs (`\t`)
- colours? (`\u001b[31m`, ... , `\u001b[0m`)

Ideally I'd love to show something that would have the class of input we need to be concerned about working with, and have it handled at least as well as the prior implementation handled it.

A specific scenario that is trivial to setup and play with, for example the newline escaped character `\n` in the prompt:

```pwsh
PS /home/d3r3kk/dev/venvtest> ../cpython/python -m venv --prompt "\nd3r3kk\n" .newline_prompt
PS /home/d3r3kk/dev/venvtest> cat ./.newline_prompt/pyvenv.cfg
home = /home/d3r3kk/dev/cpython
include-system-site-packages = false
version = 3.9.0
prompt = '\\nd3r3kk\\n'
PS /home/d3r3kk/dev/venvtest> ./.newline_prompt/bin/Activate.ps1
(\\nd3r3kk\\n) PS /home/d3r3kk/dev/venvtest> deactivate
PS /home/d3r3kk/dev/venvtest> exit
```
...and then in bash...

```bash
d3r3kk@deb:~/dev/venvtest$ . .newline_prompt/bin/activate
(
d3r3kk
) d3r3kk@deb:~/dev/venvtest$ deactivate
d3r3kk@deb:~/dev/venvtest$ 
```

...as we can see, the `\\n` being stored in the `pyvenv.cfg` is written to the prompt verbatim in Powershell with this change (and is also in the latest 3.8 beta release). 

_**Note**_: it _is_ possible in Powershell to replace some escaped character's `\\` with a ` and it will work. (But definitely not colours, those are weird special creatures!).